force-app/main/default/lwc/weatherComponent/weatherComponent.js

```diff
@@ -0,0 +1,89 @@
+import WHEATHER_KEY_FIELD from '@salesforce/schema/Account.WeatherKey__c'
+import IS_BACK_IMPL_FIELD from '@salesforce/schema/Account.IsBackimplementation__c'
+import getCurrentWeather from '@salesforce/apex/WeatherService.getCurrentWeather';
+import { getFieldValue, getRecord } from 'lightning/uiRecordApi';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import { loadStyle } from 'lightning/platformResourceLoader';
+import weather from '@salesforce/resourceUrl/weather';
+
+const fields = [SHIPPING_CITY_FIELD, WHEATHER_KEY_FIELD, IS_BACK_IMPL_FIELD];
+
+export default class WeatherComponent extends LightningElement {
+    @api recordId;
+    @track weather = {
+        temp: 0,
+        wind: 0,
+        humidity: 0
+    }
+    @wire(getRecord, {recordId: '$recordId', fields})
+    account(result){
+        if(result.data){
+            if(getFieldValue(result.data, IS_BACK_IMPL_FIELD)){
+                getCurrentWeather({location: getFieldValue(result.data, SHIPPING_CITY_FIELD)})
+                .then(value => {
+                    this.weather.temp = value.temp;
+                    this.weather.wind = value.windSpeed;
+                    this.weather.humidity = value.humidity;
+                }).catch(error => {
+                    this.dispatchEvent(
+                        new ShowToastEvent({
+                               title: 'Error!', 
+                               message: error.body.message, 
+                               variant: 'error'
+                           }),
+                      );
+                })                
+            }else{
+                fetch("https://api.openweathermap.org/data/2.5/weather?q=" + getFieldValue(result.data, SHIPPING_CITY_FIELD) + "&units=metric"  + "&appid=" + getFieldValue(result.data, WHEATHER_KEY_FIELD),
+                {
+                    method:"GET",
+                    headers:{
+                        "Accept":"application/json"
+                    }
+                })
+                .then (response => {
+                    if(response.ok){                
+                        return response.json();
+                    }
+                })
+                .then(jsonResponse => {
+                    this.weather.temp = (jsonResponse['main'].temp);
+                    this.weather.wind = jsonResponse['wind'].speed;
+                    this.weather.humidity = jsonResponse['main'].humidity;
+                    return this.weather;
+                }).catch(error => {
+                    this.dispatchEvent(
+                        new ShowToastEvent({
+                               title: 'Error!', 
+                               message: error.message, 
+                               variant: 'error'
+                           }),
+                      );   
+                });
+            }            
+            loadStyle(this, weather);
+        }else if(result.error){
```

Comment:
am not sure "else if" is require here, I mean just "else" should be enough, if result.data not equals true then result.error should equal true, please correct me if I am wrong

Reply:
I need to get access to the error that the result variable provides and then display the specific message of this result.error, that's why I use the "else if" statement